### DISCUSSION TITLE
HELP - [Kura Driver] Sequence of methods called by a Driver once Refresh button is pressed

### DISCUSSION THREAD

**Discussion by alexcorvis84** [eclipse-kura/kura#5305] [Q&A] [answered]
Hi, 

I'm developing a Kura Driver which uses [new BLE APIs](https://eclipse.github.io/kura/docs-release-5.4/java-application-development/how-to-use-new-bt-le-apis/#how-to-use-the-kura-ble-api).

I've been able to develop a (beta) working version of the driver, being able to scan for the device, read its sensor data and parse the bytes to get the sensor values, as you can see below.

![imagen](https://github.com/eclipse/kura/assets/44984326/cf1e03a7-7d1b-417e-ac2b-3843fe07b87d)

I would like to know, what is the **sequence of the methods** that are called by a Driver once the button **Refresh** is pressed, because sometimes I'm getting strange behavior of the driver.

Many thanks in advance.

**Comment by MMaiero** [comment]
Hi, 
maybe @nicolatimeus and @pierantoniomerlino can help us on this.
I would suggest to follow the call from the web ui to the asset and driver code part to get a better idea on the actions that are taken, but I would expect that the code invokes the same Api methods as the Wire graph to trigger an action on the corresponding asset and driver

**Comment by pierantoniomerlino** [comment, marked as answer]
Hi,

the Refresh button triggers a read of all the available channels in the Asset. So, it calls the `BaseAsset.readAllChannels` method [here](https://github.com/eclipse/kura/blob/71813f257ba74b032cd257e7f99d5ac2f043580d/kura/org.eclipse.kura.asset.provider/src/main/java/org/eclipse/kura/asset/provider/BaseAsset.java#L322).

**Reply by alexcorvis84** [reply]
Thanks a lot for your comment @pierantoniomerlino

**Comment by mattdibi** [comment]
Closing this as "answered". Feel free to reopen this if needed.
